include/okapi/api/coreProsAPI.hpp

```diff
@@ -94,3 +98,29 @@ class CrossplatformThread {
 
   CROSSPLATFORM_THREAD_T thread;
 };
+
+class CrossplatformMutex {
+  public:
+  CrossplatformMutex() {
```

Comment:
How can we use RAII to guarantee the lock is always released, even in the case of an unexpected exception? Just calling `unlock()` isn't enough because the method could exit early. [std::lock_guard](https://en.cppreference.com/w/cpp/thread/lock_guard) uses the constructor to take the lock and the destructor to release it. Can we do the same here? We still need an `unlock()` method though because we need to be able to release it early (i.e., before destruction).

Reply:
Turns out we actually get to use lock_guard with almost no changes.